test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -102,9 +110,427 @@ spec = do
+            let payload = Json [json| {
+                    "name": #{walName},
+                    "mnemonic_sentence": #{mnemonics24},
+                    "passphrase": "Secure Passphrase"
+                    } |]
+            r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+            verify r expectations
+
+    it "WALLETS_CREATE_04 - [] as name -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": [],
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Array"
+            ]
+
+    it "WALLETS_CREATE_04 - Num as name -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": 123,
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Number"
+            ]
+
+    it "WALLETS_CREATE_04 - Name param missing -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": "Secure Passphrase"
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "key \"name\" not present"
+            ]
+
+    describe "WALLETS_CREATE_05 - Mnemonics" $ do
+        let matrix =
+             [ ( "[] as mnemonic_sentence -> fail", []
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Invalid number of words: 15, 18, 21 or\
+                      \ 24 words are expected."
+                 ]
+               )
+             , ( "specMnemonicSentence -> fail", specMnemonicSentence
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Invalid entropy checksum: please \
+                      \double-check the last word of your mnemonic sentence."
+                 ]
+               )
+             , ( "invalid mnemonics -> fail", invalidMnemonics15
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Invalid entropy checksum: please \
+                      \double-check the last word of your mnemonic sentence."
+                 ]
+               )
+             , ( "Japanese mnemonics -> fail", japaneseMnemonics15
+               , [ expectResponseCode @IO HTTP.status400
+                 , expectErrorMessage "Found invalid (non-English) word: "
```

Comment:
@KtorZ here `Found invalid (non-English) word:` only for Japanese mnemonics, for Chinese and French throws `Invalid number of words`. Why is that?